sdk/java-sdk/src/main/scala/kalix/javasdk/impl/ProxyInfoHolder.scala

```diff
@@ -55,7 +55,10 @@ class ProxyInfoHolder(system: ExtendedActorSystem) extends Extension {
         proxyInfo.internalProxyHostname
       }
 
-    this._proxyHostname = Some(chosenProxyName)
+    // don't set if already overridden (by testkit)
+    if (this._proxyHostname.isEmpty) {
+      this._proxyHostname = Some(chosenProxyName)
+    }
```

Comment:
What about using the `def proxyHostName` to resolve it. 

So, here we do: 

```scala
this._announcedProxyHostname = 
  if (proxyInfo.internalProxyHostname.isEmpty) {
    // for backward compatibility with proxy 1.0.14 or older
    proxyInfo.proxyHostname
  } else {
    proxyInfo.internalProxyHostname
  }
```

and then 

```scala
def proxyHostname: Option[String] = 
   _proxyHostnameOverride.orElse(_announcedProxyHostname)
```

That's similar to what I did for the proxy port.

Anyway, I think both approaches are valid, but better to align it. Either like in this PR or like the proxyPort.

Reply:
I considered simplifying port to a single volatile, always feels dodgy to juggle multiple volatiles even if it probably is safe here. I'll change port to follow the same pattern as this instead.

Reply:
Done, in a safer way, that guarantee there is no race between override and discovery, in [a0026ef](https://github.com/lightbend/kalix-jvm-sdk/pull/1282/commits/a0026ef9b309d7f4ba992e120a0eef17df74d5fc)

Reply:
Heh, also, just noticed that the `_proxyAnnouncedPort` actually never was assigned in the original logic, so that can never have worked?

Reply:
> Heh, also, just noticed that the `_proxyAnnouncedPort` actually never was assigned in the original logic, so that can never have worked?

Mistake introduced by your truly here https://github.com/lightbend/kalix-jvm-sdk/pull/1273

That wasn't release yet, so we are lucky. 😄